### DISCUSSION TITLE
Changing the front page

### DISCUSSION THREAD

**Discussion by GEEOSDGs** [open-sdg/open-sdg#1657] [Q&A] [answered]
Hello, we were suggested to post our questions here.
We have a GitHub account and created the OpenSDG template. We are trying to make some configurations to the template, and we are having the following challenges:
1. Removing the front/cover page - we would like the homepage to be the first page of the SDG dashboard - we are not finding the location of the file under our GitHub account, and, we do not know how to delete it.
For example, our current website's homepage is this:
![image](https://user-images.githubusercontent.com/91685072/172700024-46dfb074-0536-4ab2-8106-17d62a35d440.png)
We would like to delete it so we go directly to the "goals page":
![image](https://user-images.githubusercontent.com/91685072/172701563-2874b15a-fbe4-421f-b142-8abbabe88520.png)

2. Accessing the "goals page" to change the title.
Currently it says "Goals":
![image](https://user-images.githubusercontent.com/91685072/172703480-20ea51b6-5c41-4e1b-a539-828112555ac6.png)

3. Configuring the targets and indicators page to two columns instead of one column, similar to how Bristol's dashboard is set up.
![image](https://user-images.githubusercontent.com/91685072/172704112-98abf549-cee7-4fce-bdc3-28b4381a6376.png)

4. Hiding the edit tab.
![image](https://user-images.githubusercontent.com/91685072/172704445-bffee774-4896-436a-8ba8-d140ee1e8fc4.png)

**Comment by brockfanning** [comment, marked as answer]
@GEEOSDGs I think I can answer some of these. I'll try to put each answer in a separate post.

Starting with the first question - how to use the "goals" layout as the front page of the site?

One of the site configurations in Open SDG is [create_pages](https://open-sdg.readthedocs.io/en/latest/configuration/#create_pages) - this setting is intended to simplify things by creating certain commonly-used pages, including the front page. The setting should be a list of pages, each of which will be created for you. For example, here is the `create_pages` that is in the "site starter" at the time of this writing:

```
create_pages:
  - filename: ''
    folder: /
    layout: frontpage-alt
    title: ''
  - filename: ''
    folder: /goals
    layout: goals
    title: ''
  - filename: ''
    folder: /reporting-status
    layout: reportingstatus
    title: ''
  - filename: indicators.json
    folder: /
    layout: indicator-json
    title: ''
  - filename: ''
    folder: /search
    layout: search
    title: ''
```

In this case the first item (the one with a "folder" of `/` and a "filename" of `'') is the front page.  If you would like to use a different layout for your front page, you could change the "layout" of this item. For example, if you want to use the "goals" layout, you could change it to:

```
create_pages:
  - filename: ''
    folder: /
    layout: goals
    title: ''
  - filename: ''
    folder: /goals
    layout: goals
    title: ''
  - filename: ''
    folder: /reporting-status
    layout: reportingstatus
    title: ''
  - filename: indicators.json
    folder: /
    layout: indicator-json
    title: ''
  - filename: ''
    folder: /search
    layout: search
    title: ''
```

So in short, the way to accomplish what you're looking for is to update your `create_pages` site configuration so that the front page is using the `goals` layout.

**Comment by brockfanning** [comment]
@GEEOSDGs I think I can answer some of these. I'll try to put each answer in a separate post.

Starting with the first question - how to use the "goals" layout as the front page of the site?

One of the site configurations in Open SDG is [create_pages](https://open-sdg.readthedocs.io/en/latest/configuration/#create_pages) - this setting is intended to simplify things by creating certain commonly-used pages, including the front page. The setting should be a list of pages, each of which will be created for you. For example, here is the `create_pages` that is in the "site starter" at the time of this writing:

```
create_pages:
  - filename: ''
    folder: /
    layout: frontpage-alt
    title: ''
  - filename: ''
    folder: /goals
    layout: goals
    title: ''
  - filename: ''
    folder: /reporting-status
    layout: reportingstatus
    title: ''
  - filename: indicators.json
    folder: /
    layout: indicator-json
    title: ''
  - filename: ''
    folder: /search
    layout: search
    title: ''
```

In this case the first item (the one with a "folder" of `/` and a "filename" of `'') is the front page.  If you would like to use a different layout for your front page, you could change the "layout" of this item. For example, if you want to use the "goals" layout, you could change it to:

```
create_pages:
  - filename: ''
    folder: /
    layout: goals
    title: ''
  - filename: ''
    folder: /goals
    layout: goals
    title: ''
  - filename: ''
    folder: /reporting-status
    layout: reportingstatus
    title: ''
  - filename: indicators.json
    folder: /
    layout: indicator-json
    title: ''
  - filename: ''
    folder: /search
    layout: search
    title: ''
```

So in short, the way to accomplish what you're looking for is to update your `create_pages` site configuration so that the front page is using the `goals` layout.

**Comment by brockfanning** [comment]
Next question - how to change the title of the "Goals" page?

This "goals" layout has a special site configuration that is used to set its title and description, which is called [goals_page](https://open-sdg.readthedocs.io/en/latest/configuration/#goals_page).

For example, to give it a title of "My goals page title" and a description of "My goals page description", you can do:

```
goals_page:
    title: My goals page title
    description: My goals page description
```

Note that -- like most configuration options -- they can use [translation keys](https://open-sdg.readthedocs.io/en/latest/glossary/#translation-keys). So if your site is multilingual you would want to use some sort of translation key, for example:

 ```
goals_page:
    title: custom.my_goals_page_title
    description: custom.my_goals_page_description
```

**Comment by brockfanning** [comment]
Next question - how to change the title of the "Goals" page?

This "goals" layout has a special site configuration that is used to set its title and description, which is called [goals_page](https://open-sdg.readthedocs.io/en/latest/configuration/#goals_page).

For example, to give it a title of "My goals page title" and a description of "My goals page description", you can do:

```
goals_page:
    title: My goals page title
    description: My goals page description
```

Note that -- like most configuration options -- they can use [translation keys](https://open-sdg.readthedocs.io/en/latest/glossary/#translation-keys). So if your site is multilingual you would want to use some sort of translation key, for example:

 ```
goals_page:
    title: custom.my_goals_page_title
    description: custom.my_goals_page_description
```

**Comment by brockfanning** [comment]
Next question - how to use a 2-column layout for the goal pages?

Prior to version 2.0.0 (which is going to be released soon) this can be done by using `goal-by-target` in your [create_goals](https://open-sdg.readthedocs.io/en/latest/configuration/#create_goals) site configuration. For example:

```
create_goals:
  layout: goal-by-target
```

However an important note here -- as of version 2.0.0, the "goal-by-target" layout will no longer be maintained by the Open SDG team. We will be maintaining only a single goal layout, which will look like the "goal-by-target-vertical" single-column layout.

The good news, though, is that it will be possible to "override" the goal layout to achieve this 2-column layout. In fact, since so many implementations prefer this 2-column layout, we will be making a blog post detailing how to override the goal layout in order to keep the 2-column layout, even after 2.0.0. So, stay tuned there, and I will try to update this answer after that happens.

**Comment by brockfanning** [comment]
Next question - how to use a 2-column layout for the goal pages?

Prior to version 2.0.0 (which is going to be released soon) this can be done by using `goal-by-target` in your [create_goals](https://open-sdg.readthedocs.io/en/latest/configuration/#create_goals) site configuration. For example:

```
create_goals:
  layout: goal-by-target
```

However an important note here -- as of version 2.0.0, the "goal-by-target" layout will no longer be maintained by the Open SDG team. We will be maintaining only a single goal layout, which will look like the "goal-by-target-vertical" single-column layout.

The good news, though, is that it will be possible to "override" the goal layout to achieve this 2-column layout. In fact, since so many implementations prefer this 2-column layout, we will be making a blog post detailing how to override the goal layout in order to keep the 2-column layout, even after 2.0.0. So, stay tuned there, and I will try to update this answer after that happens.

**Reply by Jonkat323** [reply]
Hi @brockfanning 

We would like to use the 2 column layout. Is it possible to do so with the current version?

**Reply by sarah-beasley** [reply]
Hi - there was a blog post written about using the two column layout on the Open SDG website [here](https://open-sdg.org/blog/2022-06-08-goal-by-target-override/). Please give this a go and let us know if you have any problems.

**Comment by brockfanning** [comment]
Lastly, how to hide the edit tab?

As far as I can remember there is no way to hide this tab on your [staging environment](https://open-sdg.readthedocs.io/en/latest/glossary/#environments-staging-vs-production). However, this tab will automatically be hidden on your production environment. Here are some [more details on production environments](https://open-sdg.readthedocs.io/en/latest/hosting/production/) - but please feel free to ask if anything is unclear or doesn't work as expected.

To be clear though, when I say "there is no way" that is not technically true -- there is always a way in Open SDG, because you can always "override" particular files to make customizations. The downside of overriding files is that you need to maintain your overridden version, but it is always possible. Here is a bit [more detail about overriding files](https://open-sdg.readthedocs.io/en/latest/customisation/#working-with-remote-jekyll-themes).

For example, if your goal is to remove the edit buttons, you would probably need to override the "metadata-tabs.html" file and remove [these lines](https://github.com/open-sdg/open-sdg/blob/2.1.0-dev/_includes/components/indicator/metadata-tabs.html#L29-L33). But again, I would not recommend this, and instead focus on creating a production site, where the buttons would be hidden anyway.

**Comment by brockfanning** [comment]
Lastly, how to hide the edit tab?

As far as I can remember there is no way to hide this tab on your [staging environment](https://open-sdg.readthedocs.io/en/latest/glossary/#environments-staging-vs-production). However, this tab will automatically be hidden on your production environment. Here are some [more details on production environments](https://open-sdg.readthedocs.io/en/latest/hosting/production/) - but please feel free to ask if anything is unclear or doesn't work as expected.

To be clear though, when I say "there is no way" that is not technically true -- there is always a way in Open SDG, because you can always "override" particular files to make customizations. The downside of overriding files is that you need to maintain your overridden version, but it is always possible. Here is a bit [more detail about overriding files](https://open-sdg.readthedocs.io/en/latest/customisation/#working-with-remote-jekyll-themes).

For example, if your goal is to remove the edit buttons, you would probably need to override the "metadata-tabs.html" file and remove [these lines](https://github.com/open-sdg/open-sdg/blob/2.1.0-dev/_includes/components/indicator/metadata-tabs.html#L29-L33). But again, I would not recommend this, and instead focus on creating a production site, where the buttons would be hidden anyway.
